sensorimotor/sensorimotor/TMRegion.py

```diff
@@ -0,0 +1,571 @@
+      activeApicalCells = set(numpy.where(inputs["activeColumns"] == 1)[0])
+    else:
+      activeApicalCells = None
+
+    if "formInternalConnections" in inputs:
+      formInternalConnections = inputs["formInternalConnections"]
+    else:
+      formInternalConnections = True
+    self._tm.compute(activeColumns,
+                     activeExternalCells=activeExternalCells,
+                     activeApicalCells=activeApicalCells,
+                     formInternalConnections=formInternalConnections,
+                     learn=self.learningMode)
+
+    activeCellsOutput = numpy.zeros(
+      self.getOutputElementCount("activeCells"), dtype=GetNTAReal())
+    predictedActiveCellsOutput = numpy.zeros(
+      self.getOutputElementCount("predictedActiveCells"), dtype=GetNTAReal())
+
+    activeCells = [self._tm.getCellIndex(cell) for cell in (self._tm.activeCells)]
+    activeCellsOutput[activeCells] = 1.0
+    preditedActiveCells = [self._tm.getCellIndex(cell) for cell in (self._tm.predictedActiveCells)]
+    predictedActiveCellsOutput[preditedActiveCells] = 1.0
+
+    outputs["activeCells"][:] = predictedActiveCellsOutput[:]
+    outputs["predictedActiveCells"][:] = predictedActiveCellsOutput[:]
+
+
+    # TODO: Add other outputs
+    #self._tm.activeExternalCells
+    #self._tm.activeApicalCells
+    #self._tm.unpredictedActiveColumns
+    #self._tm.predictedActiveCells
+    #self._tm.activeCells
+    #self._tm.winnerCells
+    #self._tm.activeSegments
+    #self._tm.activeApicalSegments
+    #self._tm.predictiveCells
+    #self._tm.chosenCellForColumn
+    #self._tm.matchingSegments
+    #self._tm.matchingApicalSegments
+    #self._tm.matchingCells
+
+
+  def reset(self):
+    """ Reset the state of the TM """
+    if self._tm is not None:
+      self._tm.reset()
+
+
+  def debugPlot(self, name):
+    self._tm.mmGetCellActivityPlot(activityType="activeCells",
+                                   showReset=True,
+                                   resetShading=0.75,
+                                   title="ac-{name}".format(name=name))
+    self._tm.mmGetCellActivityPlot(activityType="predictiveCells",
+                                   showReset=True,
+                                   resetShading=0.75,
+                                   title="p1-{name}".format(name=name))
+    self._tm.mmGetCellActivityPlot(activityType="predictedCells",
+                                   showReset=True,
+                                   resetShading=0.75,
+                                   title="p2-{name}".format(name=name))
+    self._tm.mmGetCellActivityPlot(activityType="predictedActiveCells",
+                                   showReset=True,
+                                   resetShading=0.75,
+                                   title="pa-{name}".format(name=name))
+
+
+  @classmethod
+  def getBaseSpec(cls):
+    """Return the base Spec for TMRegion.
+
+    Doesn't include the tm parameters
+    """
+    spec = dict(
```

Comment:
What did you mean by this? Were you referring to the fact that all of the arguments are specified in the spec rather than being parsed from the `__init__` method? Since there's no actual arguments actually specified in the `__init__` for the general temporal memories (other than `*args` and `**kwargs`) I'm not really sure what'd be the best way to populate the spec. 

`getArgumentDescriptions(tmClass.__init__)` looks through the `__init__` args so I'd be in favor of just replacing the `*args`/`**kwargs** with the actual parameter names and let that function do the work of populating the spec. Would that be a better option than manually defining the majority of the spec?